### DISCUSSION TITLE
Hibernate Reactive problem with RESTEasy Reactive

### DISCUSSION THREAD

**Discussion by SetoKaiba** [quarkusio/quarkus#37320] [Q&A] [answered]
I want to use a REST Reactive endpoint to create some data. I use the code below. I even just use empty entities. It will not return and no error printed. If I make the size less, it will sucessfully return. What's the problem?
```java
    @Path("/record/generateDataset/{recordId36}")
    @POST
    @WithTransaction
    public Uni<LiveResponse> generateDataset(JsonObject subtitle, String recordId36) {
        Uni<?> uni = Uni.createFrom().voidItem();
        Dataset dataset = new Dataset();
        uni = uni.chain(v -> dataset.persist());
        for (int i = 0; i < 3000; i++) {
                            DatasetItem datasetItem = new DatasetItem();
                            datasetItem.dataset = dataset;
                            uni = uni.chain(v -> datasetItem.persist());
        }
        return uni.chain(e -> {
            System.out.println("haha");
            return Uni.createFrom().item(new LiveResponse(LiveResponse.CODE_OK, "ok"));
        }).onFailure().invoke(e->e.printStackTrace());
    }
```

[a comment by an automated account is not shown]

**Comment by SetoKaiba** [comment]
I found out the problem. I can use Uni.combine instead to solve the problem.

**Reply by SetoKaiba** [reply]
Unmark answer because of Martin Kouba's reply.
https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Hibernate.20Reactive.20problem.20with.20RESTEasy.20Reactive/near/404267071

**Reply by SetoKaiba** [reply]
Now I use `PanacheEntityBase.persist(Iterable<?>)` instead. And, it worked. It should be without thread-safe problem now. Right? @mkouba @cescoffier @DavideD

**Reply by mkouba** [reply, marked as answer]
:+1:  Yes, that should be safe because it's using `org.hibernate.reactive.mutiny.Mutiny.Session.persistAll(Object...)` under the hood.
